query-engine/core/src/query_graph_builder/write/upsert.rs

```diff
@@ -11,13 +12,13 @@ use std::{convert::TryInto, sync::Arc};
 pub fn upsert_record(graph: &mut QueryGraph, model: ModelRef, mut field: ParsedField) -> QueryGraphBuilderResult<()> {
     graph.flag_transactional();
 
-    let where_arg: ParsedInputMap = field.arguments.lookup("where").unwrap().value.try_into()?;
+    let where_arg: ParsedInputMap = field.arguments.lookup(args::WHERE).unwrap().value.try_into()?;
 
     let filter = extract_unique_filter(where_arg, &model)?;
     let model_id = model.primary_identifier();
 
-    let create_argument = field.arguments.lookup("create").unwrap();
-    let update_argument = field.arguments.lookup("update").unwrap();
+    let create_argument = field.arguments.lookup(args::CREATE).unwrap();
```

Comment:
In this case it's valid to have create and update in args (as opposed to the nested ops comment above) and it would be totally fine to have some constants duplicated if they are semantically distinct, like args and operations.